src/components/wrapper/song-list-wrapper.tsx

```diff
@@ -0,0 +1,158 @@
+import { ChangeEvent, useEffect, useRef, useState } from "react"
+import { PlusIcon } from "@heroicons/react/24/solid"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { Suggestion } from "@/types/database-types"
+import { MagnifyingGlassCircleIcon } from "@heroicons/react/24/outline"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { useRouter } from "next/router"
+import { MagnifyingGlassIcon } from "@heroicons/react/20/solid"
+import SearchBar from "@/components/suggestion/search-bar"
+import { getRepertoireSongs, getSuggestions } from "@/services/suggestion.service"
+
+interface SongListPageWrapperProps {
+  renderSongCard: (suggestion: Suggestion) => JSX.Element
+  pageName: string
+}
+
+const SongListPageWrapper = (props: SongListPageWrapperProps) => {
+  const router = useRouter()
+  const supabaseClient = useSupabaseClient<Database>()
+  const [showSearchBar, setShowSearchBar] = useState<boolean>(false)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const [songs, setSongs] = useState<Suggestion[]>([])
+  const [showSpinner, setShowSpinner] = useState(true)
+  const [showLoadingError, setShowLoadingError] = useState(false)
+  const [searchedSong, setSearchedSong] = useState<Suggestion[]>([])
+  const [errorText, setErrorText] = useState("")
+
+  useEffect(() => {
+    setShowSpinner(true)
+    const songs =
+      props.pageName == "Suggestions"
+        ? getSuggestions(supabaseClient)
+        : getRepertoireSongs(supabaseClient)
+
+    songs
+      .then((response) => {
+        if (response.error) {
+          setShowLoadingError(true)
+          return
+        }
+
+        if (response.data?.length > 0) {
+          setSongs(response.data as Suggestion[])
+          setErrorText("")
+        } else {
+          setErrorText(
+            `Looks like there are no ${
+              props.pageName === "Suggestions" ? "Suggestions made" : "Songs in repertoire"
+            } yet! Feel free to start adding them.`
+          )
+        }
+      })
+      .catch(() => {
+        setShowLoadingError(true)
+      })
+      .finally(() => {
+        setShowSpinner(false)
+      })
+  }, [supabaseClient])
+
+  const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
+    const input = e.target.value.toLowerCase()
+    const filteredSongs = songs.filter(({ title, motivation, artist }) => {
+      return (
+        title.toLowerCase().includes(input) ||
+        motivation.toLowerCase().includes(input) ||
+        artist.some((artist) => artist.toLowerCase().includes(input))
+      )
+    })
+
+    if (filteredSongs.length === 0) {
+      setErrorText(
+        `It looks like the song you are looking for has not been added yet. Feel free to add the song!`
```

Comment:
This is not accurate for repertoire. They can't add a song to the repertoire. Perhaps just show something like "This song is currently not in the repertoire."